### PR TITLE
Logarithm axes in plots library

### PR DESCRIPTION
**Pull request by JamesxX**
(This is a continuation of changes discussed in #3, #15)

As with previously, this is a draft PR, intended to share progress and collaborate on what is a highly sought-after feature.

This implementation adds axis optios {x}-mode that takes "lin" (default) or "log", which changes logic in `transform-vec` (along with a small refactor). Additionally, a different `compute-ticks` heuristic is defined and applied, which now places major ticks on powers of `axis.base` (default: `10`), and places `n` logarithmically scaled minor ticks where `n` is `int(axis.base / ticks.minor-step)`.

image

### CODE REVIEW

File: src/plot/util.typ [hunk 1]
```diff
@@ -303,6 +303,9 @@
       axis.min -= 1; axis.max += 1
     }
 
+    axis.mode = get-axis-option(name, "mode", "lin")
+    axis.base = get-axis-option(name, "base", 10)
```
Comment (JamesxX) [inline src/plot/util.typ:307]: I imagine this could be useful for other things, like the ticks PR you have at the moment #18

File: tests/plot/ref/1.png
Comment (JamesxX) [inline tests/plot/ref/1.png:1]: I think this change is unrelated to my code? the test cases just seem to be placed differently, have visually inspected each chart and nothing jumped out

File: tests/plot/marks/test.typ
Comment (JamesxX) [inline tests/plot/marks/test.typ:1]: I'm happy for this change to be discarded, its purely stylistic and was only while I was tracking down a bug that I caused in this test case

File: src/plot/util.typ (hunk 1 above)
Comment (johannes-wolf) [reply src/plot/util.typ:307]: If we merge #18, you could pick the default formatter based on the axis options :+1:.
The formatter itself does not know about the axis.

**Review by johannes-wolf — APPROVED** [review]
LGTM.
